### PR TITLE
[Bounty] Camera Implant

### LINKED ISSUE #3689: [Bounty] $5 Camera implant
Yet another thing for security to craft and admins to 'bus.

Deliverable: A camera implant, that when implanted works like the cyborg camera, where a camera console will be able to see that person.

Cameras should be joined to the SS13 network, and start with "IMPLANT #" with some random numbers after it.
(Talked with Cenrus, this is probably better off as an admin bus only item)

### PR DESCRIPTION
**Pull request by Sarchutar**
## About The Pull Request

Bounty: https://github.com/BeeStation/BeeStation-Hornet/issues/3689
Adds a camera implant to the game which is currently only available through admin tools.
Available in VV (as shown in video) or as a spawnable item as implanter/implantcase.
(Does not follow players automatically at the moment so it kinda relies on https://github.com/BeeStation/BeeStation-Hornet/pull/3727 being merged)

Closes #3689 

<details>
<summary>
In-Game
</summary>

https://user-images.githubusercontent.com/53494785/108930152-aa7be780-7645-11eb-8d2b-af8d6bff6621.mp4

</details>

## Why It's Good For The Game

A new way for admins to spy on people.

## Changelog
:cl:
add: Camera implant which inserts a camera into you. Admin only at the moment.
/:cl:

### CODE REVIEW

File: code/game/objects/items/implants/implant_camera.dm [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/obj/item/implant/camera
+	name = "camera implant"
+	desc = "Watchful eye inside you."
+	activated = FALSE
+	var/obj/machinery/camera/camera = null
```
Comment (Archanial) [inline code/game/objects/items/implants/implant_camera.dm]:
```suggestion
	var/obj/machinery/camera/camera
```

**Comment by Sarchutar** [comment]
Gonna try to change this a bit since this doesn't work anymore with #3727.

**Comment by Sarchutar** [comment]
The camera now gets inserted directly into the mob so it works with the other pr.

**Comment by bloons3** [comment]
Satisfies bounty, contact for payment

File: code/game/objects/items/implants/implant_camera.dm [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+				<b>Name:</b> Camera Implant<BR>
+				<b>Life:</b> 24 hours after death of host<BR>
+				<b>Implant Details:</b> <BR>
+				<b>Function:</b> Remote surveillance of implanted subject."}
+	return dat
+
+/obj/item/implant/camera/on_implanted(mob/user)
+	camera = new (user)		//Insert the camera directly into the mob so the camera actually shows what it sees
+	var/rand_number = rand(1, 1000)
+	camera.c_tag = "IMPLANT #[rand_number]"
```
Comment (park66665) [inline code/game/objects/items/implants/implant_camera.dm]:
```suggestion
	camera.c_tag = "IMPLANT #[rand(1, 999)]"
```

**Comment by Sarchutar** [comment]
> Satisfies bounty, contact for payment

I will message you if/when this pr gets merged.

**Review by park66665 — APPROVED** [review]
All clear. This PR is good to go.

**Pull request merged**